File: project/app/src/main/java/org/owntracks/android/injection/modules/ActivityModules/ContactsActivityModule.java
package org.owntracks.android.injection.modules.ActivityModules;

import android.support.v7.app.AppCompatActivity;

import org.owntracks.android.injection.modules.BaseActivityModule;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.contacts.ContactsActivity;
import org.owntracks.android.ui.contacts.ContactsMvvm;
import org.owntracks.android.ui.contacts.ContactsViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseActivityModule.class)
public abstract class ContactsActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity bindActivity(ContactsActivity a);

    @Binds abstract ContactsMvvm.ViewModel bindViewModel(ContactsViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/injection/modules/android/ActivityModules/ContactsActivityModule.kt
package org.owntracks.android.injection.modules.android.ActivityModules

import androidx.appcompat.app.AppCompatActivity
import dagger.Binds
import dagger.Module
import org.owntracks.android.injection.scopes.PerActivity
import org.owntracks.android.ui.contacts.ContactsActivity
import org.owntracks.android.ui.contacts.ContactsMvvm
import org.owntracks.android.ui.contacts.ContactsViewModel

@Module(includes = [BaseActivityModule::class])
abstract class ContactsActivityModule {
    @Binds
    @PerActivity
    abstract fun bindActivity(a: ContactsActivity?): AppCompatActivity?
    @Binds
    abstract fun bindViewModel(viewModel: ContactsViewModel?): ContactsMvvm.ViewModel<*>?
}